vcenter_operator/configurator.py

```diff
@@ -316,21 +322,38 @@ def setup_os_session(self):
         )
         self.os_session = Session(auth=auth)
 
-    def _poll_nova(self):
-        if not self.os_session:
-            return
+    def _poll_nova_cells(self):
+        namespace_nova = self.global_options['namespace']
+        try:
+            secret = client.CoreV1Api().read_namespaced_secret(
+                namespace=namespace_nova,
+                name='nova-cells')
+        except client.ApiException as e:
+            LOG.error(f"Failed to get secret {namespace_nova}/nova-cells: {e}")
+            return False
+
+        if secret.type != 'Opaque':
+            LOG.error(f"Unknown secret type {secret.type} for {namespace_nova}/nova-cells. Can only work with Opaque.")
+            return False
 
         try:
-            endpoint_filter = {'service_type': 'compute', 'interface': 'public'}
-            resp = self.os_session.get('/os-cells', endpoint_filter=endpoint_filter)
-            for cell in resp.json().get('cellsv2', []):
-                self.global_options['cells'][cell['name']] = cell
-        except (HttpError, ConnectionError) as e:
-            LOG.error(f"Failed to get cells: {e}")
+            cell_names = b64decode(secret.data['cells']).split(',')
```

Comment:
I think you have to `decode()` the base64-decode result from `bytes` into `str` before you can call `split()`:

```suggestion
            cell_names = b64decode(secret.data['cells']).decode().split(',')
```

Reply:
ah never mind, you wrote your own wrapper above. missed that because it's named the same. sorry for the noise.

Reply:
Arguably you could just not have the wrapper. It's so little gain for that indirection.